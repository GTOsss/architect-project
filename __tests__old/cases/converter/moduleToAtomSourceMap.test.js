const stringifyObject = require('stringify-object');
const { sourceMapToAtom } = require('../../../src/utils/change/moduleToAtomSourceMap');
const { aliases, map } = require('./source_map/source-map-module');

const createPrettyObj = (obj) => {
  return stringifyObject(obj, {
    indent: '  ',
    singleQuotes: true,
  });
};

describe('test moduleToAtom', () => {
  test('simple', () => {
    const obj = sourceMapToAtom({
      map,
      aliases: {},
    });

    expect(createPrettyObj(obj)).toMatchSnapshot();
  });

  test('simple with aliases', () => {
    const obj = sourceMapToAtom({
      map,
      aliases,
    });

    expect(createPrettyObj(obj)).toMatchSnapshot();
  });
});
