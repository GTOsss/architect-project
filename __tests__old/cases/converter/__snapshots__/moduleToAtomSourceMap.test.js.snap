// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test moduleToAtom simple 1`] = `
"{
  map: {
    icons: [
      [
        'i',
        {
          assets: 'icons-sprite',
          test: '123',
          path: 'src/components'
        }
      ]
    ],
    index: [
      [
        'ix',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    button: [
      [
        'rc',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    textarea: [
      [
        'rc',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    input: [
      [
        'rc',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    'input-select': [
      [
        'rc',
        {
          path: 'src/components/inputs'
        }
      ]
    ]
  },
  aliases: {}
}"
`;

exports[`test moduleToAtom simple with aliases 1`] = `
"{
  map: {
    icons: [
      [
        'icons',
        {
          assets: 'icons-sprite',
          test: '123',
          path: 'src/components'
        }
      ]
    ],
    index: [
      [
        'index',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    button: [
      [
        'react-component',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    textarea: [
      [
        'react-component',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    input: [
      [
        'react-component',
        {
          path: 'src/components/inputs'
        }
      ]
    ],
    'input-select': [
      [
        'react-component',
        {
          path: 'src/components/inputs'
        }
      ]
    ]
  },
  aliases: {
    rc: 'react-component',
    s: 'store',
    ix: 'index',
    i: 'icons'
  }
}"
`;
